### PR TITLE
[output] Convert the keys of dictionaries to camelCase also

### PR DESCRIPTION
**Pull request by derekbekoe**
The dict that is passed in:
```
the_dict = {'params': {'test_key_here': 'value'}}
```

Output before:
```
  {
    "params": {
      "test_key_here": "value",
    }
  }
```

Output after:
```
  {
    "params": {
      "testKeyHere": "value",
    }
  }
```

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/_util.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def to_camelcase(s):
         return re.sub(KEYS_CAMELCASE_PATTERN, lambda x: x.group(1).upper(), s)
 
     if isinstance(obj, dict):
-        return {k: todict(v) for (k, v) in obj.items()}
+        return {to_camelcase(k): todict(v) for (k, v) in obj.items()}
```
Comment (derekbekoe) [inline src/azure-cli-core/azure/cli/core/_util.py]: `to_camelcase()` is used for the `hasattr(obj, '__dict__')` if statement below so looks like it was missed here.

**Comment by tjprescott** [comment]
In general we do not convert a dictionary key to camelCase. One example of why is tags. "this_tag" is different from "thisTag" and we don't want to automatically convert.

**Comment by tjprescott** [comment]
Actually, looking at the scope of these changes, they only impact the Role commands so @yugangw-msft should really take a look at this one.

For generic update, preserving the case of the keys for dictionaries is intentional, but this change doesn't affect that.

**Comment by derekbekoe** [comment]
If tags are represented as dictionaries then @tjprescott you're right, this change would modify tags in the way you described which was not intended as this change affects all command results that have a dict.

**Comment by tjprescott** [comment]
@derekbekoe this may be an issue then. In the generic update, 'this_tag' and 'thisTag' are both valid. The generic update logic is such that it will not convert the case for you, but one problem could be that if you show something and that output logic converts the case for you, and you then try to update that, you could get bizarre behavior:

If I have a tag called 'my_tag', and show that resource, it would display as "myTag". I would then think that is the actual key, so if I ran update with --set tags.myTag=blah, and then show again, I would have two entries for "myTag", which is not good.

**Comment by derekbekoe** [comment]
Yep @tjprescott you're right! That was not intended.
Closing the PR as the current implementation works as it should.

**Pull request closed without merging**